crates/noirc_evaluator/src/ssa/ir/function_inserter.rs

```diff
@@ -30,7 +30,7 @@ impl<'f> FunctionInserter<'f> {
     pub(crate) fn resolve(&mut self, mut value: ValueId) -> ValueId {
         value = self.function.dfg.resolve(value);
         match self.values.get(&value) {
-            Some(value) => *value,
+            Some(value) => self.resolve(*value),
```

Comment:
Since we're load results to their known results in the mem2reg pass, we can have situations now where a mapped value (e.g. the result of an array_get containing a reference) is itself mapped to another value (the value of that reference).